pkg/operator/controller/controller.go

```diff
@@ -136,10 +156,38 @@ func (r *ReconcileCDI) Reconcile(request reconcile.Request) (reconcile.Result, e
 
 	// mid delete
 	if cr.DeletionTimestamp != nil {
+		if r.isUpgrading(cr) {
+			//CDI is marked to be deleteted while upgrade flow is in process
+			reqLogger.Info("Deleting  CDI during upgrade")
+			//clenaup all resources of previous installation first
+			err := r.cleanupUnusedResources(reqLogger, cr)
+			if err != nil {
+				reqLogger.Info("Failed to cleanupUnused resource prior to CDI cr deletion during upgrade")
+				return reconcile.Result{}, nil
```

Comment:
Should be `return reconcile.Result{}, err`?

Reply:
of course, fixed